components/Mediator.vue

```diff
@@ -204,8 +204,42 @@ let resolvePermissionRequest;
 export default {
   name: 'Mediator',
   components: {AntiTrackingWizard, MediatorGreeting},
+  async created() {
+    if(window.matchMedia('(prefers-color-scheme: dark)')) {
+      this.theme = 'dark';
+    } else {
+      this.theme = 'light';
+    }
+
+    if(window.location.ancestorOrigins &&
+      window.location.ancestorOrigins.length > 0) {
+      this.relyingOrigin = window.location.ancestorOrigins[0];
+    } else {
+      const {origin} = this.$route.query;
+      this.relyingOrigin = origin;
+    }
+
+    this.relyingDomain = utils.parseUrl(this.relyingOrigin).host;
+
+    // TODO: is this the appropriate place to run this?
+    loadPolyfill(this);
+
+    // attempt to load web app manifest icon
+    const manifest = await getWebAppManifest(this.relyingDomain);
+    this.relyingOriginManifest = manifest;
```

Comment:
There's a lot of other unrelated code being added here... is your history for this branch out of date or something?

Reply:
We might not need *any* changes to this file if the other changes I recommended for default theme handling are added to `getWebAppManifestIcon`.

Reply:
Not sure to be honest. There was a merge conflict so I tried to resolve it so that is probably why it's like this.